src/imageTools/highlight.js

```diff
@@ -83,9 +88,15 @@ var cornerstoneTools = (function ($, cornerstone, cornerstoneMath, cornerstoneTo
 
         context.save();
         var data = toolData.data[0];
-
-        var selectionColor="white",
-            toolsColor="white";
+        if (!data) return;
+
+        if (pointNearTool(data, cornerstoneTools.toolCoordinates.getCoords())) {
```

Comment:
Thanks for the suggestion, I will have a look. But I remember that the 
issue was that onImageRendered wasn't called unless the mouse was near 
the handles. I will check again, using your suggestion.

Thx!

On 26-May-2015 10:01 AM, Erik Ziegler wrote:
>
> In src/imageTools/highlight.js 
> <https://github.com/chafey/cornerstoneTools/pull/44#discussion_r31008474>:
>
> > @@ -83,9 +88,15 @@ var cornerstoneTools = (function ($, cornerstone, cornerstoneMath, cornerstoneTo
> >
> >          context.save();
> >          var data = toolData.data[0];
> > -
> > -        var selectionColor="white",
> > -            toolsColor="white";
> > +        if (!data) return;
> > +
> > +        if (pointNearTool(data, cornerstoneTools.toolCoordinates.getCoords())) {
>
> You can use the pointInsideRect function to check if the cursor is 
> inside the highlight tool.
>
> Something like this should work:
>
> var  coords=  cornerstoneTools.toolCoordinates.getCoords();
> if  (pointNearTool(data, coords)||  pointInsideRect(data, coords))
> {
> ...
> }...
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/chafey/cornerstoneTools/pull/44/files#r31008474>.
>